R/SEIRD_RU.R

```diff
@@ -0,0 +1,642 @@
+          # rate of change: rural
+          dsy <- - (b * sy * (iu + iy) * (f_urban * N_U + f_rural * N_Y) * C +
+                    b * sy * (iy / f_rural) * N_Y * (1 - C))
+          dey <- b * sy * (iu + iy) * (f_urban * N_U + f_rural * N_Y) * C +
+                 b * sy * (iy / f_rural) * N_Y * (1 - C) - k * ey
+          diy <- k * ey - (g + m) * iy
+          dry <- g * iy
+          dcy <- b * sy * (iu + iy) * (f_urban * N_U + f_rural * N_Y) * C +
+                 b * sy * (iy / f_rural) * N_Y * (1 - C)
+          d_deathy <- m * iy
+          # return the rate of change
+          list(c(dsu, deu, diu, dru, dcu, d_deathu,
+                 dsy, dey, diy, dry, dcy, d_deathy))
+        })
+    }
+
+    # call ode solver
+    out <- ode(
+      y = state, times = times, func = right_hand_side,
+      parms = parameters, method = solve_method)
+
+    output <- as.data.frame.array(out)
+
+    # Compute incidences and deaths: urban
+    output$CU[2:length(output$CU)] <- output$CU[
+      2:length(output$CU)] - output$CU[1:(length(output$CU) - 1)]
+    output$CU[1] <- 0
+    output$DU[2:length(output$DU)] <- output$DU[
+      2:length(output$DU)] - output$DU[1:(length(output$DU) - 1)]
+
+    # Compute incidences and deaths: rural
+    output$CY[2:length(output$CY)] <- output$CY[
+      2:length(output$CY)] - output$CY[1:(length(output$CY) - 1)]
+    output$CY[1] <- 0
+    output$DY[2:length(output$DY)] <- output$DY[
+      2:length(output$DY)] - output$DY[1:(length(output$DY) - 1)]
+
+    colnames(output) <- c("time", object@output_names)
+
+    # Create long format of output
+    output <- melt(output, id.vars = "time")
+    names(output) <- c("time", "compartment", "value")
+
+    # Added for consistency of output format across models
+    output$age_group <- rep("0-150", length(output$time))
+
+    return(output)
+  })
+
+#----------------------------------------------------------------------------
+#' @describeIn SEIRD_RU Returns the value of R0, the basic reproduction number,
+#' for the model with the specified parameter values.
+#' R0 is computed using the next generation matrix method as set out in 
+#' van den Driessche (2017):
+#' van  den  Driessche, P. Reproduction numbers of infectious disease models.
+#' Infectious Disease Modelling, 2:288–303, 8 2017.
+#' 
+#' In this method, the variables are split into two groups: those that represent
+#' infected individuals (in our case the exposed and infectious compartments in
+#' both communities), and those that do not (the susceptible, recovered and dead).
+#' We only further consider the system of equations consisting of the ODEs 
+#' describing the variables in the first group,
+#' in this case (E_U, E_Y, I_U and I_Y).
+#' The differential equations for the variables in the first group are written
+#' in the form
+#' \deqn{\frac{dx_i}{dt} = F_i - V_i}
+#' where F_i are the terms describing how individuals enter the
+#' compartment and V_i the terms describing how individuals leave.
```

Comment:
Thanks for adding the extra information here.
I plan to actually check your code, but right now I'm still learning the method. I have a few questions up front---
- The paper says that the derivatives should be evaluated at the disease free equilibrium, but it looks like you are evaluating at the actual initial conditions of the model. Is this allowed?
- The paper you cite says that "V_i(x) is the rate of other transitions between compartment i and other
**infected** compartments." But in most other descriptions of the method, and your implementation, V_i seems to contain transitions to non-infected compartments as well, such as R. Do you have any idea what they mean by this?
- `where F_i are the terms describing how individuals enter the compartment and V_i the terms describing how individuals leave.` I think we can specify that, in general, F_i contains only those individuals entering the compartment that are considered new infections. As for V_i, however my previous question is answered, it seems that it's not restricted to individuals leaving but can contain other transitions.

Reply:
In response to your points:
1) Yes, you are correct, we need to be at a disease-free equilibrium. I have fixed this now. Not sure why I used the initial conditions before, probably because they are very, very close to the disease-free equilibrium. Thanks for catching this!
2) I have simply followed the example in the paper. If you want to do it differently, please let me know what maths you want changed and why. I am happy with how it is now.
3) done.